apps/web-connect/src/features/checkout/components/ActionSection.tsx

```diff
@@ -25,20 +25,10 @@ export function ActionSection(): JSX.Element {
         mintWithXai,
         mintWithEthError,
         approve,
-        getApproveButtonText
+        getApproveButtonText,
+        getEthButtonText,
     } = useWebBuyKeysContext();
 
-    /**
-     * Determines the text to display on the main action button for ETH transactions
-     * 
-     * @returns {string} The button text
-     */
-    const getButtonText = useCallback(() => {
-        if (mintWithEth.isLoading || mintWithXai.isLoading) return "WAITING FOR CONFIRMATION";
-        if (chain?.id !== 42161) return "Please Switch to Arbitrum One";
-        return "BUY NOW";
-    }, [mintWithEth.isLoading, mintWithXai.isLoading, chain]);
-
```

Comment:
Make sure to check this when deploying the sepolia version, this will have to be updated on the sepolia branch.